applications/MappingApplication/custom_utilities/mapper.h

```diff
@@ -182,15 +182,15 @@ class Mapper
 
     // Constructor, can only be called by derived classes (actual mappers)
     Mapper(ModelPart& rModelPartOrigin, ModelPart& rModelPartDestination,
-           Parameters JsonParameters) :
```

Comment:
Hi Philip,

the behaviour of parameters is "strange" in the sense that the Parameters object holds internally a pointer to the data of the lib it is built upon. This means that the correct way of passing it is

     Parameters JsonParameters

which will behave much like as if it was a reference. 
Passing a reference is actually a memory error, so i must block your pull request for this.

The point is that the wrapping of the rapidjson library was a little tricky, as we needed to ensure that the data exists until the last Parameters object exists. This is the reason for which we hold a shared ptr to the underlying data (with the counter being updated when you pass a Parameters object)